### ISSUE TITLE
[Bug] New-DbaAvailabilityGroup - Missing check for HADR beeing enabled

### ISSUE THREAD

**Issue by andreasjordan** [sqlcollaborative/dbatools#6727]
I accidentally ran New-DbaAvailabilityGroup without Enable-DbaAgHadr before, so I got this verbose output:

```
WARNING: [09:29:16][Get-DbaAvailabilityGroup] Availability Group (HADR) is not configured for the instance: SRV1\SQL2016.
VERBOSE: Performing the operation "Setting up availability group named TestAg and adding primary replica" on target "SRV1\SQL2016".
VERBOSE: [09:29:17][Get-DbaEndpoint] Getting endpoint Dedicated Admin Connection on SRV1\SQL2016
VERBOSE: [09:29:17][Get-DbaEndpoint] Getting endpoint TSQL Default TCP on SRV1\SQL2016
VERBOSE: [09:29:17][Get-DbaEndpoint] Getting endpoint TSQL Default VIA on SRV1\SQL2016
VERBOSE: [09:29:17][Get-DbaEndpoint] Getting endpoint TSQL Local Machine on SRV1\SQL2016
VERBOSE: [09:29:17][Get-DbaEndpoint] Getting endpoint TSQL Named Pipes on SRV1\SQL2016
VERBOSE: [09:29:19][Get-DbaEndpoint] Getting endpoint hadr_endpoint on SRV1\SQL2016
VERBOSE: [09:29:21][Get-DbaLogin] Getting last login times
VERBOSE: [09:29:22][New-DbaLogin] Adding as login type WindowsUser
VERBOSE: [09:29:22][New-DbaLogin] Successfully added COMPANY\SQLServer to SRV1\SQL2016.
VERBOSE: [09:29:22][Get-DbaLogin] Getting last login times
VERBOSE: [09:29:22][Get-DbaLogin] Processing [COMPANY\SQLServer] on SRV1\SQL2016
WARNING: [09:29:22][Get-DbaAvailabilityGroup] Availability Group (HADR) is not configured for the instance: SRV1\SQL2016.
VERBOSE: [09:29:22][Get-DbaLogin] Getting last login times
VERBOSE: [09:29:22][Get-DbaLogin] Processing [COMPANY\SQLServer] on SRV1\SQL2016
VERBOSE: Performing the operation "Adding cluster permissions for availability group named TestAg" on target "SRV1\SQL2016".
VERBOSE: [09:29:22][New-DbaAvailabilityGroup] WSFC Cluster requires granting [NT AUTHORITY\SYSTEM] a few things. Setting now.
VERBOSE: Performing the operation "Adding replica to availability group named TestAg" on target "SRV2\SQL2016".
VERBOSE: [09:29:22][Get-DbaEndpoint] Getting endpoint Dedicated Admin Connection on SRV2\SQL2016
VERBOSE: [09:29:22][Get-DbaEndpoint] Getting endpoint TSQL Default TCP on SRV2\SQL2016
VERBOSE: [09:29:22][Get-DbaEndpoint] Getting endpoint TSQL Default VIA on SRV2\SQL2016
VERBOSE: [09:29:22][Get-DbaEndpoint] Getting endpoint TSQL Local Machine on SRV2\SQL2016
VERBOSE: [09:29:22][Get-DbaEndpoint] Getting endpoint TSQL Named Pipes on SRV2\SQL2016
VERBOSE: [09:29:24][Get-DbaEndpoint] Getting endpoint hadr_endpoint on SRV2\SQL2016
VERBOSE: [09:29:26][Get-DbaLogin] Getting last login times
VERBOSE: [09:29:26][New-DbaLogin] Adding as login type WindowsUser
VERBOSE: [09:29:26][New-DbaLogin] Successfully added COMPANY\SQLServer to SRV2\SQL2016.
VERBOSE: [09:29:26][Get-DbaLogin] Getting last login times
VERBOSE: [09:29:26][Get-DbaLogin] Processing [COMPANY\SQLServer] on SRV2\SQL2016
WARNING: [09:29:26][Get-DbaAvailabilityGroup] Availability Group (HADR) is not configured for the instance: SRV2\SQL2016.
VERBOSE: [09:29:26][Get-DbaLogin] Getting last login times
VERBOSE: [09:29:26][Get-DbaLogin] Processing [COMPANY\SQLServer] on SRV2\SQL2016
WARNING: [09:29:27][New-DbaAvailabilityGroup] An exception occurred while executing a Transact-SQL statement or batch. | Could not process the operation. Always On Availability Groups replica manager is dis
abled on this instance of SQL Server. Enable Always On Availability Groups, by using the SQL Server Configuration Manager. Then, restart the SQL Server service, and retry the currently operation. For inform
ation about how to enable and disable Always On Availability Groups, see SQL Server Books Online.
```

The WARNING in the first line is from this code in New-DbaAvailabilityGroup:
https://github.com/sqlcollaborative/dbatools/blob/5e988826581f05cbfcc281bbbea52930a325ef72/functions/New-DbaAvailabilityGroup.ps1#L306-L309

The code in Get-DbaAvailabilityGroup:
https://github.com/sqlcollaborative/dbatools/blob/5e988826581f05cbfcc281bbbea52930a325ef72/functions/Get-DbaAvailabilityGroup.ps1#L85-L87

So best would be to check .IsHadrEnabled in New-DbaAvailabilityGroup before using Get-DbaAvailabilityGroup and do one of two things:
1. Stop the command with the warning as in Get-DbaAvailabilityGroup (and maybe point the user to the command Enable-DbaAgHadr)
2. Try to run Enable-DbaAgHadr to enable HADR. But that would not work in every situation because of permissions and would restart the instance.

Before I start to code this I want to have an agreement on the way we want it to work.

**Comment by wsmelton** [comment]
Option one only. The command scope is creating the AG and should not be enabling configuration options for an instance level setting.

**Comment by andreasjordan** [comment]
> Option one only. The command scope is creating the AG and should not be enabling configuration options for an instance level setting.

Ok, I agree with you. Should I open a pull request for that or should we wait for others to vote?

**Comment by wsmelton** [comment]
Fix we need to use would be to change the call to the get function to have EnableException and wrap it in a try catch block.

When we make calls to public functions it should be doing this so the errors are thrown properly.

That way we also don't have to repeat code that is already being done.

**Comment by andreasjordan** [comment]
> Fix we need to use would be to change the call to the get function to have EnableException and wrap it in a try catch block.
> 
> When we make calls to public functions it should be doing this so the errors are thrown properly.
> 
> That way we also don't have to repeat code that is already being done.

I disagree here. The only duplicate would be `if (-not $server.IsHadrEnabled) {` as I would rephrase the error message to point the user to Enable-DbaAgHadr. And for reading and maintaining the code a clear test for a requirement is better than hiding that in the test for existence of the AG.

So you should provide the code for this issue.
